scripts/docs/snippetadapter.js

```diff
@@ -7,83 +7,244 @@
+ * @param {Object} options
+ * @param {Boolean} options.production Whether to build snippets in production mode.
+ * @param {Array.<String>>|undefined} options.whitelistedSnippets An array that contains glob patterns.
+ * @param {Object.<String, Function>} umbertoHelpers
+ * @returns {Promise}
+ */
+module.exports = function snippetAdapter( snippets, options, umbertoHelpers ) {
+	const { getSnippetPlaceholder, getSnippetSourcePaths } = umbertoHelpers;
+	const snippetsDependencies = new Map();
+
+	// For each snippet, load its config. If the snippet has defined dependencies, load those as well.
+	for ( const snippetData of snippets ) {
+		if ( !snippetData.snippetSources.js ) {
+			throw new Error( `Missing snippet source for "${ snippetData.snippetName }".` );
+		}
+
+		snippetData.snippetConfig = readSnippetConfig( snippetData.snippetSources.js );
+		snippetData.snippetConfig.language = snippetData.snippetConfig.language || DEFAULT_LANGUAGE;
+
+		if ( snippetData.snippetConfig.dependencies ) {
+			for ( const dependencyName of snippetData.snippetConfig.dependencies ) {
+				// Do not load the same dependency more than once.
+				if ( snippetsDependencies.has( dependencyName ) ) {
+					continue;
+				}
+
+				// Find a root path where to look for the snippet's sources. We just want to pass it through Webpack.
+				const snippetBasePathRegExp = new RegExp( snippetData.snippetName.replace( /\//g, '\\/' ) + '.*$' );
+				const snippetBasePath = snippetData.snippetSources.js.replace( snippetBasePathRegExp, '' );
+
+				const dependencySnippet = {
+					snippetSources: getSnippetSourcePaths( snippetBasePath, dependencyName ),
+					snippetName: dependencyName,
+					outputPath: snippetData.outputPath,
+					destinationPath: snippetData.destinationPath,
+					requiredFor: snippetData
+				};
+
+				if ( !dependencySnippet.snippetSources.js ) {
+					throw new Error( `Missing snippet source for "${ dependencySnippet.snippetName }".` );
+				}
+
+				dependencySnippet.snippetConfig = readSnippetConfig( dependencySnippet.snippetSources.js );
+				dependencySnippet.snippetConfig.language = dependencySnippet.snippetConfig.language || DEFAULT_LANGUAGE;
+
+				snippetsDependencies.set( dependencyName, dependencySnippet );
+			}
+		}
 	}
 
-	const snippetConfig = readSnippetConfig( data.snippetSource.js );
-	const outputPath = path.join( data.outputPath, data.snippetPath );
+	// Add all dependencies to the snippet collection.
+	for ( const snippetData of snippetsDependencies.values() ) {
+		snippets.add( snippetData );
+	}
 
-	const webpackConfig = getWebpackConfig( {
-		entry: data.snippetSource.js,
-		outputPath,
-		language: snippetConfig.language,
-		production: data.options.production,
-		definitions: data.options.definitions || {}
-	} );
+	// Remove snippets that do not match to patterns specified in `options.whitelistedSnippets`.
+	if ( options.whitelistedSnippets ) {
```

Comment:
Now it's present all the time.

Reply:
Or I might be wrong as this file will be executed by the bigbangdocs...

Reply:
In that case I'm marking this as resolved.